core/metaobjectrepository.cpp

```diff
@@ -183,6 +185,39 @@ void MetaObjectRepository::initIOTypes()
     MO_ADD_METAOBJECT1(QSaveFile, QFileDevice);
 }
 
+Q_DECLARE_METATYPE(const QObject*)
+Q_DECLARE_METATYPE(void**)
+
+void MetaObjectRepository::initQEventTypes()
+{
+    MetaObject *mo = nullptr;
+    MO_ADD_METAOBJECT0(QEvent);
+    MO_ADD_PROPERTY_RO(QEvent, isAccepted);
+    MO_ADD_PROPERTY_RO(QEvent, spontaneous);
+    MO_ADD_PROPERTY_RO(QEvent, type);
+
+    MO_ADD_METAOBJECT1(QTimerEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QTimerEvent, timerId);
+
+    MO_ADD_METAOBJECT1(QChildEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QChildEvent, child);
+    MO_ADD_PROPERTY_RO(QChildEvent, added);
+    MO_ADD_PROPERTY_RO(QChildEvent, polished);
+    MO_ADD_PROPERTY_RO(QChildEvent, removed);
+
+    MO_ADD_METAOBJECT1(QDynamicPropertyChangeEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QDynamicPropertyChangeEvent, propertyName);
+
+    MO_ADD_METAOBJECT1(QDeferredDeleteEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QDeferredDeleteEvent, loopLevel);
+
+    MO_ADD_METAOBJECT1(QMetaCallEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QMetaCallEvent, id);
+//    MO_ADD_PROPERTY_RO(QMetaCallEvent, sender);  // problematic because type is const QObject*
```

Comment:
There is precedent for this, see "const QStyle* QStyle::proxy() const" as supported by the widget plugin. This property is also shown correctly in the property model (and recursively expanded).